### ISSUE TITLE
cop_distinct applies to entire set of columns, not the only column specified in the argument

### ISSUE THREAD

**Issue by pavelkveton** [qorelanguage/qore#812]
take the following example:

$table.selectRows(("columns": (cop_distinct("name"), "id"), "where": ("type": "user")));

at least for Oracle, in this case you get list of distinct tuples "name","id". 

I.e. having the table like
name   id
A          0
A          1
A          0

then the above statement returns 2 rows
A   0
A   1

Which is what I would expect acc. to my understanding of distinct in sql.

The problem is that cop_distinct applied on single column is VERY confusing.

Could we make "distinct" a flag/switch in the select hash, something like:

$table.selectRows(("distinct" : True, "columns": ("name", "id"), "where": ("type": "user")));

**Comment by gamato** [comment]
No, that would not work as there are other use cases (e.g. `select count (distinct id)`).

**Comment by davidnich** [comment]
removed the "module" label since that is only meant for binary modules and not for Qore user modules
